packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx

```diff
@@ -74,8 +74,8 @@ export default function NavbarContent(): JSX.Element {
         <>
           <NavbarItems items={rightItems} />
           <NavbarColorModeToggle className={styles.colorModeToggle} />
-          {autoAddSearchBar && (
-            <NavbarSearch>
+          {searchBarItem && (
```

Comment:
Ah, wait, my bad. The logic should be the other way: `!searchBarItem && (...)`. This is when the user doesn't specify a search item explicitly and we "auto add" it.

Reply:
Oh totally true, I missed that part. Nice catch!

Reply:
Then I got the logic wrong anyways. Updated in a fixup commit